BruTile/TileIndex.cs

```diff
@@ -8,13 +8,25 @@ public struct TileIndex : IComparable
     {
         public int Col { get; }
         public int Row { get; }
-        public string Level { get; }  // Note: TileIndex is a struct but Level is a class which needs GC. It would be nice if we could avoid GC.
+        public int Zoom { get; private set; }
+        public string Level     // Note: TileIndex is a struct but Level is a class which needs GC. It would be nice if we could avoid GC.

```

Comment:
Agree with avoiding GC. 

The problem with the parse solution is that WMTS specification officially allows strings to indicate levels. So you can not always parse it. I found this very annoying when I read that the first time. If fixed this by making Level type string, but I regret that. What I should have done is work with an internal dictionary holds the mapping from our integer Level to the wmts string value. In our api we should work with the integer level while in our wmts implementation we should translate from string to int.

When parsing we should do a sort on the resolution of the level to make sure that our integer levels are in the correct order.

Reply:
> What I should have done is work with an internal dictionary that holds the mapping from our integer Level to the wmts string value. In our api we should work with the integer level while in our wmts implementation we should translate from string to int.
> When parsing we should do a sort on the resolution of the level to make sure that our integer levels are in the correct order.

I 2nd that 👍